students/ethan_nguyen/lesson03/mailroom.py

```diff
@@ -0,0 +1,87 @@
+
+import operator
+'''
+Class to store Donor object and its attributes
+'''
+class Donor:
+    def __init__(self, name, amount=0, num=0):
+        self.name = name
+        self.amount_gift = amount
+        self.num_gift = num
+    '''function to calculate average donation'''
+    def getAverge(self):
+        return (self.amount_gift/self.num_gift)
+    '''function to return number of donations'''
+    def getCount(self):
+        return self.num_gift
+    '''function to return historical amount'''
+    def getAmount(self):
+        return self.amount_gift
+    def getName(self):
+        return self.name
+    '''function to add new donation amount to historical'''
+    def addAmount(self, addition):
+        self.amount_gift += addition
+        self.num_gift += 1
+
+if __name__ == "__main__":
+    #create a dict object to store donors
+    DonorDict = {}
```

Comment:
variable names are snake_case -> `donor_dict`

also, at this point in the lesson your data structure should be list/tuple combo or just a list

Reply:
Hi Natasha,

Thank you for your feedback.  I will update mailroom.py to use list/tuple combo.  I was following the instructions on https://startlearning.uw.edu/courses and it doesn't have the more detailed instructions as https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom-part1.html.   It looks like we should follow the exercises posted on the latter link? 

thanks,

Reply:
I was referring to these guidelines on the course website:
```
Guidelines
First, factor your script into separate functions. Each of the above tasks can be accomplished by a series of steps. Write discreet functions that accomplish individual steps and call them.

Second, use loops to control the logical flow of your program. Interactive programs are a classic use-case for the while loop.

Of course, input() will be useful here.

Put the functions you write into the script at the top.

Put your main interaction into an if __name__ == '__main__' block.

Finally, use only functions and the basic Python data types you’ve learned about so far. There is no need to go any farther than that for this assignment.```


yes you can refer to github pages for more detailed information 🙂